python/semantic_kernel/connectors/memory/postgres/postgres_collection.py

```diff
@@ -360,38 +417,167 @@ async def _create_index(self, table_name: str, vector_field: VectorStoreRecordVe
-                """
-                SELECT table_name
-                FROM information_schema.tables
-                WHERE table_schema = %s AND table_name = %s
-                """,
-                (self.db_schema, self.collection_name),
+            await cur.execute(query, params)
+            rows = await cur.fetchall()
+            # Add the distance to the results
+            results = [convert_row_to_dict(row, return_fields) for row in rows]
+            return KernelSearchResults(
+                results=self._get_vector_search_results_from_results(results, options),
+                total_count=len(results) if options.include_total_count else None,
             )
-            row = await cur.fetchone()
-            return bool(row)
 
-    @override
-    async def delete_collection(self, **kwargs: Any) -> None:
-        """Delete the collection."""
-        if self.connection_pool is None:
-            raise VectorStoreOperationException(
-                "Connection pool is not available, use the collection as a context manager."
+    def _construct_vector_query(
+        self,
+        vector: list[float | int],
+        fields: list[tuple[str, VectorStoreRecordField]],
+        options: VectorSearchOptions,
+        **kwargs: Any,
+    ) -> tuple[sql.Composed, list[Any], list[tuple[str, VectorStoreRecordField | None]]]:
+        """Construct a vector search query.
+
+        Args:
+            vector: The vector to search for.
+            fields: The fields.
+            options: The search options.
+            **kwargs: Additional arguments.
+
+        Returns:
+            The query, parameters, and the fields representing the columns in the result.
+        """
+        # Get the vector field we will be searching against,
+        # defaulting to the first vector field if not specified
+        vector_fields = self.data_model_definition.vector_fields
+        if not vector_fields:
+            raise VectorSearchExecutionException("No vector fields defined.")
+        if options.vector_field_name:
+            vector_field = next((f for f in vector_fields if f.name == options.vector_field_name), None)
+            if not vector_field:
+                raise VectorSearchExecutionException(f"Vector field '{options.vector_field_name}' not found.")
+        else:
+            vector_field = vector_fields[0]
+
+        # Default to cosine distance if not set
+        distance_function = vector_field.distance_function or DistanceFunction.COSINE_DISTANCE
+        ops_str = get_vector_distance_ops_str(distance_function)
+
+        # Select all fields except all vector fields if include_vectors is False
+        select_fields = [(name, f) for (name, f) in fields if (name != vector_field.name or options.include_vectors)]
+        select_list = [name for (name, _) in select_fields]
+
+        where_clause = self._build_where_clauses_from_filter(options.filter)
+
+        query = sql.SQL("SELECT {}, {} {} %s as {} FROM {}.{}").format(
```

Comment:
nit: I find a f-string easier to read when there are this many fields being filled... or something like this: `"{arg1} {arg2}".format(arg1=10, arg2=20)`

Reply:
Thanks, I've changed all sql.SQL.format calls to use named parameters in PostgresCollection.